Makefile

```diff
@@ -801,13 +801,13 @@ docker-build-kubeadm-control-plane: ## Build the docker image for kubeadm contro
 
 .PHONY: docker-build-docker-infrastructure
 docker-build-docker-infrastructure: ## Build the docker image for docker infrastructure controller manager
```

Comment:
WDYT about making this change to all our docker builds? (not just the ones we know we had issues with)

Reply:
Let's also add a short comment inline why we do this (pointing to the issue or something)

Reply:
this was my logic for leaving the rest of the build commands untouched:
> All of the unmodified `docker-build-*` targets leverage the `--build-arg` package option and intentionally use the same Dockerfile located in the root dir

from my understanding they are intentionally using the same Dockerfile so the fact that it is cached is not an issue. happy to make the change if you'd like though

Reply:
Ah got it. I would prefer implementing it for all. I'm pretty sure there is a chance we won't remember this issue when we ever have to move one of these to another Dockerfile. It's also more consistent (not that it's super important :))

Reply:
(sorry for not reading the PR description)

Reply:
np!  will go ahead and make these changes

Reply:
pushed up the updates when you have a chance to take another look @sbueringer